Tools/run_batch.py

```
@@ -106,11 +122,120 @@ def _parse_run_config(path: Path) -> run_config_pb2.RunConfig:
 
+def _validate_link_config(link_config, label: str) -> None:
+    """Rejects invalid link values before any worker processes are launched."""
+    if (not math.isfinite(link_config.latency_seconds) or
+            link_config.latency_seconds < 0):
+        raise ValueError(f"{label} latency_seconds must be non-negative.")
+    if (not math.isfinite(link_config.latency_std_seconds) or
+            link_config.latency_std_seconds < 0):
+        raise ValueError(f"{label} latency_std_seconds must be non-negative.")
+    if (not math.isfinite(link_config.packet_delivery_ratio) or
+            not 0 <= link_config.packet_delivery_ratio <= 1):
+        raise ValueError(f"{label} packet_delivery_ratio must be in [0, 1].")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)